unittests/UnitTestFramework.xml

```diff
@@ -261,12 +341,56 @@ begin
 				errorline = p.linenumber,
 				errorexec =	p.Name);
 		endif;
+	elseif expectedString is not null then
+		if equalStrings(s1=expectedString, s2=actualString)=TRUE then
+			if CurObject.TraceStringsOnFailure=TRUE then
+				CurMethod.Trace(text = HC_NEWLINE+'=== Actual/expected string (equal): ===');
+				CurMethod.Trace(string = actualString);
+			endif;
+			CurObject.throwAssertionFailedError(errortext=errortext+HC_NEWLINE+HC_TAB+
+				'assertNotEquals: actualString is equal expectedString',
+				errorline = p.linenumber,
+				errorexec =	p.Name);
+		endif;
+	elseif expectedJSON is not null then
+		if equalJSON(json1=expectedJSON, json2=actualJSON)=TRUE then
+			if CurObject.TraceStringsOnFailure=TRUE then
+				CurMethod.Trace(text = HC_NEWLINE+'=== Actual/expected JSON (equal): ===');
+				CurMethod.Trace(string = actualJSON);
+			endif;
+			CurObject.throwAssertionFailedError(errortext=errortext+HC_NEWLINE+HC_TAB+
+				'assertNotEquals: actualJSON is equal expectedJSON',
+				errorline = p.linenumber,
+				errorexec =	p.Name);
+		endif;
+	elseif expectedJsonValue is not null then
+		if actualJsonValue is not null then
+			jh = CurSession.JsonHandler;
+			j1 = stringobject.Create();
+			j2 = stringobject.Create();
+			jh.Write(value=expectedJsonValue, json=j1, sortobjectmembers=TRUE);
+			jh.Write(value=actualJsonValue, json=j2, sortobjectmembers=TRUE);

```

Comment:
Could you add indented=TRUE

Reply:
See my comment for previous line.

Reply:
This is a fair statement.  I was thinking how it would visually look not about the real purpose which is comparison.  I will approve it as it is.